docs/spec/spec.md

```diff
@@ -0,0 +1,309 @@
+# Object Model
+
+This is the detailed specification of the Knative Eventing Object Model. For
```

Comment:
I think we may want to pivot this document slightly, and focus on the contracts. Under that guise, you might rewrite this as:

```markdown
Knative ships with several objects which implement useful sets of [interfaces](interfaces.md). It is expected that additional objects which implement other sets of interfaces will be added over time. For context, see the [overview](overview.md) and [motivations](motivation.md) sections.
```